src/coreclr/interop/interoplib.cpp

```diff
@@ -129,7 +128,22 @@ namespace InteropLib
                     checkForIdentity = trackerObject.p;
             }
 
-            return checkForIdentity->QueryInterface(identity);
+            HRESULT hr;
+
+            IUnknown* identityLocal;
+            RETURN_IF_FAILED(checkForIdentity->QueryInterface(&identityLocal));
+
+            // Set the inner if scenario dictates an update.
+            if (*innerMaybe == nullptr          // User didn't supply inner - legacy scenario sanity check.
+                && checkForIdentity != external // Target of check was changed - legacy scenario sanity check.
+                && external != identityLocal    // The supplied object doesn't match the computed identity.
+                && refTrackerInnerScenario)     // The appropriate flags were set.
+            {
+                *innerMaybe = external;
+            }
```

Comment:
Just to confirm, the "legacy scenario" here is aggregation+ref tracker support without the new API, correct?

Reply:
Yep. Change phrasing?

Reply:
Yeah it would be good to mention it a bit more specifically before the if block.